pkg/neorpc/errors.go

```diff
@@ -30,33 +30,95 @@ const (
+	// ErrUnknownHeader is returned when requested header is not found.
+	ErrUnknownHeader = NewError(RPCErrorUnknownHeaderCode, "Unknown header", "")
+
+	// ErrInvalidHeight represents SubmitError with code -201.
+	ErrInvalidHeight = NewSubmitError(-201, "Invalid height")
+	// ErrInvalidVerificationFunction represents SubmitError with code -202.
+	ErrInvalidVerificationFunction = NewSubmitError(-202, "Invalid verification function")
+
+	// ErrUnknown represents SubmitError with code -500.
+	ErrUnknown = NewSubmitError(-500, "Unclassified inventory verification error")
 	// ErrAlreadyExists represents SubmitError with code -501.
-	ErrAlreadyExists = NewSubmitError(-501, "Block or transaction already exists and cannot be sent repeatedly.")
+	ErrAlreadyExists = NewSubmitError(-501, "Inventory already exists on chain")
 	// ErrOutOfMemory represents SubmitError with code -502.
-	ErrOutOfMemory = NewSubmitError(-502, "The memory pool is full and no more transactions can be sent.")
+	ErrOutOfMemory = NewSubmitError(-502, "The memory pool is full and no more transactions can be sent")
 	// ErrUnableToVerify represents SubmitError with code -503.
-	ErrUnableToVerify = NewSubmitError(-503, "The block cannot be validated.")
+	ErrUnableToVerify = NewSubmitError(-503, "Transaction already exists in the memory pool")
 	// ErrValidationFailed represents SubmitError with code -504.
-	ErrValidationFailed = NewSubmitError(-504, "Block or transaction validation failed.")
+	ErrValidationFailed = NewSubmitError(-504, "Validation failed")
 	// ErrPolicyFail represents SubmitError with code -505.
-	ErrPolicyFail = NewSubmitError(-505, "One of the Policy filters failed.")
-	// ErrUnknown represents SubmitError with code -500.
-	ErrUnknown = NewSubmitError(-500, "Unknown error.")
+	ErrPolicyFail = NewSubmitError(-505, "One of the Policy filters failed")
+	// ErrInvalidScript represents SubmitError with code -506.
+	ErrInvalidScript = NewSubmitError(-506, "Invalid script")
+	// ErrTransactionAttr represents SubmitError with code -507.
+	ErrTransactionAttr = NewSubmitError(-507, "Invalid transaction attribute")
+	// ErrInvalidSignature represents SubmitError with code -508.
+	ErrInvalidSignature = NewSubmitError(-508, "Invalid signature")
+	// ErrInvalidSize represents SubmitError with code -509.
+	ErrInvalidSize = NewSubmitError(-509, "Invalid inventory size")
+	// ErrExpiredTransaction represents SubmitError with code -510.
+	ErrExpiredTransaction = NewSubmitError(-510, "Expired transaction")
+	// ErrInsufficientFunds represents SubmitError with code -511.
+	ErrInsufficientFunds = NewSubmitError(-511, "Insufficient funds")
+
+	// ErrOracleNotRun represents SubmitError with code -601.
+	ErrOracleNotRun = NewSubmitError(-601, "Oracle service is not running")
+	// ErrOracleFinished represents SubmitError with code -602.
+	ErrOracleFinished = NewSubmitError(-602, "Oracle request has already been finished")
+	// ErrOracleNotFound represents SubmitError with code -603.
```

Comment:
`SubmitError` doesn't mean a lot, we better have descriptions explaining semantics from the proposal.

Reply:
The same note about helper, see the https://github.com/nspcc-dev/neo-go/pull/3063#discussion_r1269625976. We may make it unexported to use it only from neorpc package.

Reply:
Not exactly, what I mean here is that descriptions should provide more useful data to our dear users. If you're to rename the helper and have
```
	// ErrOracleNotFound represents ErrorWithCode with code -603.
```
it won't be a lot better wrt this.

Reply:
Fixed